src/plugins/profiling/common/profiling.ts

```diff
@@ -31,75 +29,90 @@ export type StackFrame = {
   SourceType: number;
 };
 
+export type Executable = {
+  FileName: string;
+};
+
 export type StackFrameMetadata = {
+  // StackTrace.FileID
   FileID: FileID;
-  AddressOrLine: number;
+  // StackTrace.Type
+  FrameType: number;
+  // stringified FrameType -- FrameType.String()
+  FrameTypeString: string;
 
+  // StackFrame.LineNumber?
+  AddressOrLine: number;
```

Comment:
Do we need numbers here at all (or: are we going to do calculations with the number variables) ? We can possibly use strings everywhere and avoid a lot of conversions.

Reply:
Even though I couldn't find an instance where the field is used in calculations, the downstream consumer of the flamegraph (PIxi flamechart) expects `AddressOrLine` to be a number. For type safety and consistency, my preference is to keep the `number` type so that any issues with data integrity are highlighted earlier.

With that said, it would be useful as we get closer to MVP to determine how much converting between types and graph structures impacts the overall memory usage and latency.